packages/shared/lib/core/app/actions/downloadAppUpdate.ts

```diff
@@ -9,61 +9,36 @@ import {
 import { NotificationData } from '@lib/typings/notification'
 import { Platform } from '@lib/platform'
 
-import { installAppUpdate } from '../../utils'
+import { installAppUpdate } from '../utils'
 import {
     appUpdateBusy,
     appUpdateComplete,
     appUpdateError,
     appUpdateMinutesRemaining,
     appUpdateProgress,
-} from '../../stores'
+} from '../stores'
 import { cancelAppUpdateDownload } from './cancelAppUpdateDownload'
+import { Unsubscriber } from 'svelte/store'
 
-// TODO: Look for `appUpdateDownload` throughout the project
-export function downloadAppUpdate(): void {
+function resetAppUpdateStores(): void {
     appUpdateProgress.set(0)
     appUpdateMinutesRemaining.set(-1)
     appUpdateBusy.set(true)
     appUpdateComplete.set(false)
     appUpdateError.set(false)
+}
 
-    let progressSubscription = null
-    let minutesRemainingSubscription = null
-    let completeSubscription = null
-    let errorSubscription = null
-
-    const cleanup = () => {
-        removeDisplayNotification(notificationId)
-        progressSubscription()
-        completeSubscription()
-        errorSubscription()
-        minutesRemainingSubscription()
-    }
-
-    const downloadingNotification: NotificationData = {
-        type: 'info',
-        message: localize('notifications.downloadingUpdate'),
-        progress: 0,
-        subMessage: localize('notifications.calcMinutesRemaining'),
-        actions: [
-            {
-                label: localize('actions.cancel'),
-                callback: () => {
-                    cancelAppUpdateDownload()
-                    cleanup()
-                },
-            },
-        ],
-        timeout: NOTIFICATION_TIMEOUT_NEVER,
-    }
-
-    const notificationId = showAppNotification(downloadingNotification)
+/**
+ * Initializes the download for an application update.
+ */
+export function downloadAppUpdate(): void {
+    resetAppUpdateStores()
 
-    progressSubscription = appUpdateProgress.subscribe((progress) => {
+    const progressSubscription: Unsubscriber = appUpdateProgress.subscribe((progress) => {
```

Comment:
I like your solution here, but you don't need to type the variables here. `subscribe()` can only return an `Unsubscriber`, so the type will be automatically inferred.